src/components/fileStorage/repo/files.repo.js

```diff
@@ -0,0 +1,52 @@
+const { FileModel } = require('./db');
+
+const searchQuery = (userId) => ({
+	permissions: {
+		$elemMatch: {
+			refId: userId,
+		}
+	}
+});
+
+const getFileById = async (id) => FileModel.findById(id).lean().exec();
+
+/**
+ * @param {*} userId
+ * @return {data} filePermissions
+ */
+const getFilesWithUserPermissionsByUserId = async (userId) => {
+	return FileModel.aggregate([
+		{
+			$match: searchQuery(userId),
+		},
+		{
+			$project: {
+				_id: 1,
+				permissions: {
+					$filter: {
+						input: '$permissions',
+						as: 'permission',
+						cond: { $eq: ['$$permission.refId', userId] },
+					},
+				},
+			},
+		},
+	]);
+};
+
+/**
+ * @param {*} userId
+ * @return {boolean} success
+ */
+const removeFilePermissionsByUserId = async (userId) => {
+	const updateQuery = { $pull: { permissions: { refId: userId } } };
```

Comment:
Any deletion and modfied operations must validate if userId is not undefined or null.